core/src/main/java/ch/cyberduck/core/http/CustomHttpRequestExecutor.java

```diff
@@ -77,8 +87,28 @@ protected HttpResponse doSendRequest(final HttpRequest request, final HttpClient
         }
         final HttpResponse response = super.doSendRequest(request, conn, context);
         if(null != response) {
-            // response received as part of an expect-continue handshake
+            // Response received as part of an expect-continue handshake
             this.log(response);
+            if(new HostPreferences(host).getBoolean("request.unauthorized.ntlm.preflight")) {
+                // Unable to authenticate with NTLM for requests with non-zero entity
+                if(HttpStatus.SC_UNAUTHORIZED == response.getStatusLine().getStatusCode()) {
+                    if(response.containsHeader(HttpHeaders.WWW_AUTHENTICATE)) {
+                        switch(request.getRequestLine().getMethod()) {
```

Comment:
Also add `POST` to make it more generic for non-WebDAV HTTP protocols supporting NTLM?